### PR TITLE
scroll to top button component created and implemented in home page

### LINKED ISSUE #410: Feature request: Add "ToTop" button on the home page.
On the Home page of the doc2pen website add a "to top" button in the bottom right corner of the page

### PR DESCRIPTION
**Pull request by ashuvssut**
updated:
image

closes #410

- I have **used a react-bootstrap** button 
- made it into a **reusable react component** so that it can be used in all pages when needed
- button only appear when users has scrolled down more than 50px
- smooth-scroll when clicked on button

-- **(DWoC and SWoC Participant)**

### CODE REVIEW

**Comment by ashuvssut** [comment]
@anushbhatia. Sir, a deployment error showed up.
```
remote: Permission to smaranjitghose/doc2pen.git denied to github-actions[bot].

fatal: unable to access 'https://github.com/smaranjitghose/doc2pen/': The requested URL returned error: 403

Error: Process completed with exit code 128.
```
Is there some fault in my code or is this some bug. Can you please tell me about this. I will try to avoid that mistake in my next PRs.

**Comment by smaranjitghose** [comment]
> @anushbhatia. Sir, a deployment error showed up.
> 
> ```
> remote: Permission to smaranjitghose/doc2pen.git denied to github-actions[bot].
> 
> fatal: unable to access 'https://github.com/smaranjitghose/doc2pen/': The requested URL returned error: 403
> 
> Error: Process completed with exit code 128.
> ```
> 
> Is there some fault in my code or is this some bug. Can you please tell me about this. I will try to avoid that mistake in my next PRs.
> right now, code in my pr works alright

When does this error come up?

**Comment by ashuvssut** [comment]
> > @anushbhatia. Sir, a deployment error showed up.
> > ```
> > remote: Permission to smaranjitghose/doc2pen.git denied to github-actions[bot].
> > 
> > fatal: unable to access 'https://github.com/smaranjitghose/doc2pen/': The requested URL returned error: 403
> > 
> > Error: Process completed with exit code 128.
> > ```
> > 
> > 
> > Is there some fault in my code or is this some bug. Can you please tell me about this. I will try to avoid that mistake in my next PRs.
> > right now, code in my pr works alright
> 
> When does this error come up?

In some of my previous PRs also this same error showed up. Even now I can see there are other PRs too that have this error. But the code works all fine

**Comment by ashuvssut** [comment]
sir, I have an important observation about this error.

ref #388. sir this PR #388 doesnt show this error coz it never had that 7th check: the build check.
whereas my PR and some other PRs have undergone this 7th check and showed this same error.

I am sure enough that my code works completely fine. I think this 7th check always fails. I am still unsure though😅

**Review by anushbhatia — CHANGES_REQUESTED** [review]
Don't use material UI for now on this we can implement in this later stage of the project currently you can use bootstrap to create the component for top button through it.

**Comment by ashuvssut** [comment]
i will push changes today itself

**Comment by anushbhatia** [comment]
Sure

**Comment by smaranjitghose** [comment]
Try to do a squash commit

**Comment by ashuvssut** [comment]
@anushbhatia please add DWoC label also😊

**Comment by ashuvssut** [comment]
updated my pr description. U can see the new image.

**Review by anushbhatia — CHANGES_REQUESTED** [review]
Screenshot 2021-01-22 124139
Shadow is pretty weird and the button is not completely centre aligned.

File: src/components/ScrollToTopButton/style.css [hunk 1]
```diff
@@ -0,0 +1,18 @@
+.to-top-button {
+  z-index: 10000;
+  position: fixed !important;
+  bottom: 1rem;
+  right: 1rem;
+  background-color: white !important;
+  box-shadow: 0 0 6px rgba(0, 0, 0, 0.3);
```
Comment (anushbhatia) [inline src/components/ScrollToTopButton/style.css:7]: The spread of the shadow is the weird cause of colour I think

Comment (ashuvssut) [reply src/components/ScrollToTopButton/style.css]: image

**Pull request merged**